cargo/cargo_build_script_runner/bin.rs

```diff
@@ -31,121 +31,182 @@ fn main() -> Result<(), String> {
-                "" => {
-                    const SYS_CRATE_SUFFIX: &str = "-sys";
-                    if crate_name.ends_with(SYS_CRATE_SUFFIX) {
-                        crate_name
-                            .split_at(crate_name.rfind(SYS_CRATE_SUFFIX).unwrap())
-                            .0
-                    } else {
-                        &crate_name
-                    }
-                },
-                crate_links => crate_links,
-            };
-
-            write(&envfile, BuildScriptOutput::to_env(&output, &exec_root.to_string_lossy()).as_bytes())
-                .expect(&format!("Unable to write file {:?}", envfile));
-            write(&depenvfile, BuildScriptOutput::to_dep_env(&output, crate_links, &exec_root.to_string_lossy()).as_bytes())
-                .expect(&format!("Unable to write file {:?}", depenvfile));
-
-            let CompileAndLinkFlags { compile_flags, link_flags } = BuildScriptOutput::to_flags(&output, &exec_root.to_string_lossy());
-
-            write(&flagfile, compile_flags.as_bytes())
-                .expect(&format!("Unable to write file {:?}", flagfile));
-            write(&linkflags, link_flags.as_bytes())
-                .expect(&format!("Unable to write file {:?}", linkflags));
-            Ok(())
+    let CompileAndLinkFlags {
+        compile_flags,
+        link_flags,
+    } = BuildScriptOutput::to_flags(&output, &exec_root.to_string_lossy());
+
+    write(&flagfile, compile_flags.as_bytes())
+        .expect(&format!("Unable to write file {:?}", flagfile));
+    write(&linkflags, link_flags.as_bytes())
+        .expect(&format!("Unable to write file {:?}", linkflags));
+    Ok(())
+}
+
+/// A representation of expected command line arguments.
+struct Options {
+    progname: String,
+    crate_name: String,
+    crate_links: String,
+    out_dir: String,
+    envfile: String,
+    flagfile: String,
+    linkflags: String,
+    output_dep_env_path: String,
+    input_dep_env_paths: Vec<String>,
+}
+
+/// Parses positional comamnd line arguments into a well defined struct
+fn parse_args() -> Result<Options, String> {
+    let mut args = env::args().skip(1);
+
+    // TODO: we should consider an alternative to positional arguments.
+    match (args.next(), args.next(), args.next(), args.next(), args.next(), args.next(), args.next(), args.next()) {
+        (
+            Some(progname), 
+            Some(crate_name), 
+            Some(crate_links), 
+            Some(out_dir), 
+            Some(envfile), 
+            Some(flagfile), 
+            Some(linkflags), 
+            Some(output_dep_env_path)
+        ) => {
+            Ok(Options{
+                progname,
+                crate_name,
+                crate_links,
+                out_dir,
+                envfile,
+                flagfile,
+                linkflags,
+                output_dep_env_path,
+                input_dep_env_paths: args.collect();
```

Comment:
```suggestion
                input_dep_env_paths: args.collect(),
```

Reply:
woah, how did the tests pass? 😓 
```
//cargo/cargo_build_script_runner:test                          (cached) PASSED in 0.2s
//test:greeting_rule_test                                       (cached) PASSED in 0.3s
//test:hello_lib_rule_test                                      (cached) PASSED in 0.1s
//test:hello_world_rule_test                                    (cached) PASSED in 0.4s
//test/build_env:arbitrary_env_test                             (cached) PASSED in 0.4s
//test/build_env:conflicting_deps_test                          (cached) PASSED in 0.2s
//test/chained_direct_deps:mod1_test                            (cached) PASSED in 0.2s
//test/chained_direct_deps:mod2_test                            (cached) PASSED in 0.0s
//test/chained_direct_deps:mod3_doc_test                        (cached) PASSED in 0.7s
//test/chained_direct_deps:mod3_test                            (cached) PASSED in 0.4s
//test/clippy:bad_test                                          (cached) PASSED in 0.4s
//test/clippy:ok_test                                           (cached) PASSED in 0.3s
//test/conflicting_deps:conflicting_deps_test                   (cached) PASSED in 0.3s
//test/inline_test_with_deps/test:inline_test                   (cached) PASSED in 0.5s
//test/inline_test_with_deps/test_with_srcs:inline_test         (cached) PASSED in 0.4s
//test/process_wrapper:arg_files_test                           (cached) PASSED in 0.5s
//test/process_wrapper:basic_test                               (cached) PASSED in 0.1s
//test/process_wrapper:combined_test                            (cached) PASSED in 0.5s
//test/process_wrapper:copy_output_test                         (cached) PASSED in 0.1s
//test/process_wrapper:env_files_test                           (cached) PASSED in 0.1s
//test/process_wrapper:stderr_test                              (cached) PASSED in 0.2s
//test/process_wrapper:stdout_test                              (cached) PASSED in 0.2s
//test/process_wrapper:subst_pwd_test                           (cached) PASSED in 0.1s
//test/renamed_deps:mod1_test                                   (cached) PASSED in 0.4s
//test/renamed_deps:mod2_test                                   (cached) PASSED in 0.4s
//test/renamed_deps:mod3_test                                   (cached) PASSED in 0.3s
//test/rust_test_binary:wrapped_rust_test                       (cached) PASSED in 0.2s
//test/rustfmt:rustfmt_test                                     (cached) PASSED in 0.4s
//test/test_env:test                                            (cached) PASSED in 0.6s
//tools/runfiles:runfiles_doc_test                              (cached) PASSED in 0.7s
//tools/runfiles:runfiles_test                                  (cached) PASSED in 0.5s
```

This has been updated.

Reply:
I guess because that file is not covered by these tests.

Reply:
Yeah, these don't depend on any build scripts :)